cpp/src/arrow/array-list-test.cc

```diff
@@ -313,4 +445,21 @@ TEST_F(TestListArray, TestBuilderPreserveFieleName) {
   ASSERT_EQ("counts", type.value_field()->name());
 }
 
+TEST_F(TestLargeListArray, TestBuilderPreserveFieleName) {
```

Comment:
Do we really want to copy/paste all ListArray tests instead of using some kind of parametrization?